src/TwigComponent/src/ComponentFactory.php

```diff
@@ -184,7 +184,8 @@ private function preMount(object $component, array $data, ComponentMetadata $com
             $newData = $component->{$method->name}($data);
 
             if (null !== $newData) {
-                $data = $newData;
+                // merge data/options so that extra data is not lost between hooks
+                $data = $newData + $data;
```

Comment:
How do you guys emphasis/warn to not write something for futur devs?

We should never simplify `$data = $newData + $data` to `$data += $newData` as it will produce unexpected behavior

Reply:
I wonder if we should use `$data = array_merge($data, $newData)`. Maybe it's just me, but this is more explicit that using the `+` operator (as this operator always trips me up).

Reply:
@Kocal pointed it out for _using less opcodes_
https://github.com/symfony/ux/issues/1704#issuecomment-2046929715

Reply:
Heh, I did read that too. Let's keep using + then. Will the test fail if the sides are switched? That's the best way to ensure this doesn't get changed in the future.

Reply:
> Will the test fail if the sides are switched?

Yes, that's why I've added the `PreMount` inside the test. 
And another one: `Symfony\UX\TwigComponent\Tests\Integration\ComponentExtensionTest::testCanRenderTheSameComponentMultipleTimes`

> That's the best way to ensure this doesn't get changed in the future.

Some will think they need to change the tests, no?

Reply:
> Yes, that's why I've added the PreMount inside the test.

Ok, perfect!

> Some will think they need to change the tests, no?

I'm not concerned. The note and tests are all we can do.